File: CPP/ByRound/1721/1721_C_Min_Max_Array_Transformation.cpp
// Problem: CF 1721 C - Min-Max Array Transformation
// https://codeforces.com/contest/1721/problem/C

/*
Algorithm/Techniques: 
- For each element in the array `a`, we determine the minimum and maximum possible values of `d[i]` such that when we add `d[i]` to `a[i]` and sort the result, we get the array `b`.
- Minimum `d[i]`: For the i-th element in `a`, we find the smallest element in `b` that is greater than or equal to `a[i]`, and the difference between them becomes the minimum `d[i]`.
- Maximum `d[i]`: We track an index `ind` that moves forward to determine the maximum `d[i]` such that `a[i] + d[i] <= b[ind]` and the resulting sorted array `b` is valid.

Time Complexity: O(n log n) per test case due to sorting and binary search.
Space Complexity: O(n) for storing the arrays `a` and `b`.
*/

#include <algorithm>
#include <iostream>

using namespace std;
int a[1000005], b[1000005], n;
int main() {
  int t;
  cin >> t;
  while (t--) {
    cin >> n;
    for (int i = 0; i < n; i++)
      cin >> a[i];
    for (int i = 0; i < n; i++)
      cin >> b[i];
    
    // Calculate minimum d[i] values
    for (int i = 0; i < n; i++)
      cout << *lower_bound(b, b + n, a[i]) - a[i] << ' '; // Find the smallest b[j] >= a[i]
    cout << endl;
    
    // Calculate maximum d[i] values
    int ind = 0;
    for (int i = 0; i < n; i++) {
      ind = max(ind, i); // Make sure ind doesn't go backward
      // Move ind forward while a[ind + 1] <= b[ind]
      while (ind < n - 1 && a[ind + 1] <= b[ind])
        ind++;
      cout << b[ind] - a[i] << ' '; // Max d[i] is b[ind] - a[i]
    }
    cout << endl;
  }
}


// https://github.com/VaHiX/CodeForces/

File: Python/ByTier/C/1721_C_Min_Max_Array_Transformation.py
# Problem: CF 1721 C - Min-Max Array Transformation
# https://codeforces.com/contest/1721/problem/C

"""
Algorithm: Min-Max Array Transformation
Techniques: Two Pointers, Greedy Approach

Time Complexity: O(n) per test case
Space Complexity: O(n) for storing arrays and results

This problem involves finding the minimum and maximum values of d_i such that 
when we add d_i to a_i, the resulting array b is sorted in non-descending order.
We use a greedy two-pointer approach to efficiently determine these bounds.
"""

import sys

input = sys.stdin.readline
ans = []
for _ in range(int(input())):
    n = int(input())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    c = [0] * n  # To store minimum values of d_i
    d = [0] * n  # To store maximum values of d_i
    p, q = n - 1, n - 1  # Two pointers for tracking the positions in b
    
    # Traverse from right to left
    for i in range(n - 1, -1, -1):
        # Move pointer p to the left as long as b[p-1] >= a[i]
        # This ensures that we find the smallest valid element in b for current a[i]
        while p > 0 and b[p - 1] >= a[i]:
            p -= 1
        # Minimum d_i is the difference between current element in b and a[i]
        c[i] = b[p] - a[i]
        # Maximum d_i is the difference between the latest valid element in b and a[i]
        d[i] = b[q] - a[i]
        # Update q pointer if p reaches i, meaning we have used up the element at p
        if p == i:
            q = i - 1
            
    ans.append(c)
    ans.append(d)

for i in ans:
    print(*i)


# https://github.com/VaHiX/CodeForces/